Reachable unwrap in sys_rt_sigprocmask() caused by Bad user space pointer specified

### Describe the bug

There is a reachable unwrap caused by Bad user space pointer specified in `sys_rt_sigprocmask()` at kernel/aster-nix/src/syscall/rt_sigprocmask.rs:29 when make a `rt_sigprocmask` syscall with specific arguments.
https://github.com/asterinas/asterinas/blob/0291b5dc6bb142b9c6165c1cb29b7658eefdaa63/kernel/aster-nix/src/syscall/rt_sigprocmask.rs#L29-L30


### To Reproduce
1. Compile a program which calls `rt_sigprocmask` with specific arguments:
```C
#include <stdio.h>
#include <sys/syscall.h>
#include <unistd.h>
#include <signal.h>
#include <string.h>
#include <errno.h>

int main() {
    syscall(SYS_rt_sigprocmask, 0x0, 0x8, 0x9, 128);

    return 0;
}
```
2. Run the compiled program in Asterinas.

### Expected behavior
Asterinas reports panic and is terminated.

### Environment
- Official docker asterinas/asterinas:0.6.2
- 11th Gen Intel(R) Core(TM) i7-1165G7 @ 2.80GHz
- Asterinas version: main 0291b5dc

### Logs
```
~ # /root/rt_sigprocmask-test.c 
[    11.721] ERROR: sigset size is not equal to 8
panicked at /root/asterinas/kernel/aster-nix/src/syscall/rt_sigprocmask.rs:29:58:
called `Result::unwrap()` on an `Err` value: Error { errno: EFAULT, msg: Some("Bad user space pointer specified") }
Printing stack trace:
   1: fn 0xffffffff887cc8c0 - pc 0xffffffff887cc8d8 / registers:

     rax               0x12; rdx 0xffffffff889e5000; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007e6aa560;

   2: fn 0xffffffff887cc6a0 - pc 0xffffffff887cc818 / registers:

     rax               0x12; rdx 0xffffffff889e5000; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007e6aa570;

   3: fn 0xffffffff88049000 - pc 0xffffffff8804900a / registers:

     rax               0x12; rdx 0xffffffff889e5000; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007e6aa6f0;

   4: fn 0xffffffff8895b780 - pc 0xffffffff8895b802 / registers:

     rax               0x12; rdx 0xffffffff889e5000; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007e6aa700;

   5: fn 0xffffffff88938ab0 - pc 0xffffffff88938bdb / registers:

     rax               0x12; rdx 0xffffffff889e5000; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007e6aa790;

   6: fn 0xffffffff884809b0 - pc 0xffffffff88480d7d / registers:

     rax               0x12; rdx 0xffffffff889e5000; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007e6aa890;

   7: fn 0xffffffff883b1520 - pc 0xffffffff883b78d0 / registers:

     rax               0x12; rdx 0xffffffff889e5000; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007e6aab40;

   8: fn 0xffffffff88352cf0 - pc 0xffffffff88352d7e / registers:

     rax               0x12; rdx 0xffffffff889e5000; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007e6c03d0;

   9: fn 0xffffffff88446ab0 - pc 0xffffffff8844762f / registers:

     rax               0x12; rdx 0xffffffff889e5000; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007e6c0570;

  10: fn 0xffffffff8808cc00 - pc 0xffffffff8808cc0e / registers:

     rax               0x12; rdx 0xffffffff889e5000; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007e6c0f90;

  11: fn 0xffffffff887a6ce0 - pc 0xffffffff887a6cf6 / registers:

     rax               0x12; rdx 0xffffffff889e5000; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007e6c0fb0;

  12: fn 0xffffffff888435b0 - pc 0xffffffff88843619 / registers:

     rax               0x12; rdx 0xffffffff889e5000; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007e6c0fd0;

  13: fn                0x0 - pc                0x0 / registers:

     rax               0x12; rdx 0xffffffff889e5000; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff80007e6c1000;

[OSDK] The kernel seems panicked. Parsing stack trace for source lines:
(  1) /root/asterinas/ostd/src/panicking.rs:107
(  2) /root/asterinas/ostd/src/panicking.rs:59
(  3) a3wadqd69fx2whpv190t0ri68:?
(  4) ??:?
(  5) ??:?
(  6) /root/.rustup/toolchains/nightly-2024-06-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/result.rs:1102
(  7) /root/asterinas/kernel/aster-nix/src/syscall/mod.rs:173
(  8) /root/asterinas/kernel/aster-nix/src/syscall/mod.rs:323
(  9) /root/asterinas/kernel/aster-nix/src/thread/task.rs:69
( 10) /root/.rustup/toolchains/nightly-2024-06-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ops/function.rs:79
( 11) /root/.rustup/toolchains/nightly-2024-06-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/boxed.rs:2077
( 12) /root/asterinas/ostd/src/task/task/mod.rs:310
make: *** [Makefile:153: run] Error 1
```

--- a/kernel/aster-nix/src/syscall/futex.rs
+++ b/kernel/aster-nix/src/syscall/futex.rs
@@ -19,7 +19,7 @@ pub fn sys_futex(
     ctx: &Context,
 ) -> Result<SyscallReturn> {
     // FIXME: we current ignore futex flags
-    let (futex_op, futex_flags) = futex_op_and_flags_from_u32(futex_op as _).unwrap();
+    let (futex_op, futex_flags) = futex_op_and_flags_from_u32(futex_op as _)?;
     debug!(
         "futex_op = {:?}, futex_flags = {:?}, futex_addr = 0x{:x}",
         futex_op, futex_flags, futex_addr
@@ -42,24 +42,24 @@ pub fn sys_futex(
 
     let res = match futex_op {
         FutexOp::FUTEX_WAIT => {
-            let timeout = get_futex_timeout(utime_addr).expect("Invalid time addr");
+            let timeout = get_futex_timeout(utime_addr)?;
             futex_wait(futex_addr as _, futex_val as _, &timeout).map(|_| 0)
         }
         FutexOp::FUTEX_WAIT_BITSET => {
-            let timeout = get_futex_timeout(utime_addr).expect("Invalid time addr");
+            let timeout = get_futex_timeout(utime_addr)?;
             futex_wait_bitset(futex_addr as _, futex_val as _, &timeout, bitset as _).map(|_| 0)
         }
         FutexOp::FUTEX_WAKE => {
-            let max_count = get_futex_val(futex_val as i32).expect("Invalid futex val");
+            let max_count = get_futex_val(futex_val as i32)?;
             futex_wake(futex_addr as _, max_count).map(|count| count as isize)
         }
         FutexOp::FUTEX_WAKE_BITSET => {
-            let max_count = get_futex_val(futex_val as i32).expect("Invalid futex val");
+            let max_count = get_futex_val(futex_val as i32)?;
             futex_wake_bitset(futex_addr as _, max_count, bitset as _).map(|count| count as isize)
         }
         FutexOp::FUTEX_REQUEUE => {
-            let max_nwakes = get_futex_val(futex_val as i32).expect("Invalid futex val");
-            let max_nrequeues = get_futex_val(utime_addr as i32).expect("Invalid utime addr");
+            let max_nwakes = get_futex_val(futex_val as i32)?;
+            let max_nrequeues = get_futex_val(utime_addr as i32)?;
             futex_requeue(
                 futex_addr as _,
                 max_nwakes,

--- a/kernel/aster-nix/src/syscall/futex.rs
+++ b/kernel/aster-nix/src/syscall/futex.rs
@@ -69,8 +69,7 @@ pub fn sys_futex(
             .map(|nwakes| nwakes as _)
         }
         _ => panic!("Unsupported futex operations"),
-    }
-    .unwrap();
+    }?;
 
     debug!("futex returns, tid= {} ", ctx.thread.tid());
     Ok(SyscallReturn::Return(res as _))

--- a/kernel/aster-nix/src/syscall/rt_sigprocmask.rs
+++ b/kernel/aster-nix/src/syscall/rt_sigprocmask.rs
@@ -26,7 +26,7 @@ pub fn sys_rt_sigprocmask(
     if sigset_size != 8 {
         error!("sigset size is not equal to 8");
     }
-    do_rt_sigprocmask(mask_op, set_ptr, oldset_ptr, ctx).unwrap();
+    do_rt_sigprocmask(mask_op, set_ptr, oldset_ptr, ctx)?;
     Ok(SyscallReturn::Return(0))
 }
 

